appserver/node-express/lib/db-client/qnaDoc/index.js

```diff
@@ -22,6 +22,18 @@ funcs.search = function (spec) {
   ).result();
 };
 
+/**
+ * Get one or more documents from the Samplestack database based ID(s).
+ * For single documents, apply a transform to include voting and
+ * reputation information.
+ *
+ * @param  {String} txid The transaction ID.
+ * @param  {Object|Array} spec An object with a document ID property. Example:
+ *   {"id":"b6ff5e47-2d08-4c20-9a16-4998fbb1bc10"}
+ * For multiple documents, an array of objects (TODO Is this correct? Where is
```

Comment:
There are a few reasons it is this way.

1. I think we may hit the multiple document scenario in configuring the database and/or testing
2. In many different promise chains, we need to know how many results we got, and being able to say in the error message that we expected 1 and got, for instance, 3, is of value, so that function (cardinatity check) is built into the getUnique generalized function. That's why in this particular function we don't go the distance to make sure that there's only one -- it's generalized elsewhere (in part because of #1).

Anyway, it could have gone a few different directions, the design as is leaves the code more generalized so that as features are added which require getting more than one document, this function can remain in service (especially if its method of turning the spec into criteria for the client api is improved,which it should be).